subsys/bluetooth/controller/hci/hci.c

```diff
@@ -6628,7 +6628,11 @@ static void le_scan_req_received(struct pdu_data *pdu_data,
 static void le_conn_complete(struct pdu_data *pdu_data, uint16_t handle,
 			     struct net_buf *buf)
 {
-	struct node_rx_cc *cc = (void *)pdu_data;
+	void *node = (void *)pdu_data;
+
+	LL_ASSERT(IS_PTR_ALIGNED(node, struct node_rx_cc));
+
+	struct node_rx_cc *cc = node;
```

Comment:
This: https://github.com/zephyrproject-rtos/zephyr/blob/aa83b8d6e3bb5e9fbe35912350e9faa6c6e9c5b2/subsys/bluetooth/controller/ll_sw/lll.h#L391-L401

should have avoided misaligned access?

Reply:
No, because `node_rx_cc` and `node_rx_pdu` are not packed and have alignment requirements, whereas `struct pdu_data` is packed and thus has zero alignment requirements. Hence, the compiler complains.

Reply:
`__aligned(4)` ensures that inside `strict node_rx_pdu` the field `pdu` is  not unaligned. The allocation by `mem_acquire` ensures runtime aligned memory for each node.

>  because node_rx_cc and node_rx_pdu are not packed and have alignment requirements, whereas struct pdu_data is packed and thus has zero alignment requirements. Hence, the compiler complains.

Agree; on the compile time complain.